code/messaging/src/androidTest/java/com/adobe/marketing/mobile/messaging/E2EFunctionalTests.java

```diff
@@ -145,50 +163,147 @@ Map<String, Object> createExpectedEdgePersonalizationEventData() {
+        assertEquals(
+                expectedRuleConsequenceDetails.get("data"),
+                triggeredRuleConsequenceDetails.get("data"));
+        assertEquals(
+                expectedRulesConsequenceEventData.get("type"),
+                triggeredConsequenceEventData.get("type"));
+    }
 
+    @Test
+    public void testIamTriggerShowOnce() throws InterruptedException {
         // test
-        Messaging.refreshInAppMessages();
+        verifyInAppPropositionsRetrievedFromEdge();
 
-        // verify messaging request content event from refreshInAppMessages API call
-        final List<Event> messagingRequestEvents =
-                getDispatchedEventsWith(EventType.MESSAGING, EventSource.REQUEST_CONTENT);
-        assertEquals(1, messagingRequestEvents.size());
-        final Event messagingRequestEvent = messagingRequestEvents.get(0);
-        assertEquals(true, messagingRequestEvent.getEventData().get("refreshmessages"));
+        // trigger a show once in-app message
+        MobileCore.trackAction("once", null);
+
+        // verify show once rule consequence event is dispatched
+        final Map<String, Object> expectedRulesConsequenceEventData =
+                (Map<String, Object>)
+                        ((List)
+                                        MessagingTestUtils.getMapFromFile(
+                                                        "iam-show-once-consequence.json")
+                                                .get(JSON_CONSEQUENCES_KEY))
+                                .get(0);
+        List<Event> rulesConsequenceEvents =
+                getDispatchedEventsWith(EventType.RULES_ENGINE, EventSource.RESPONSE_CONTENT);
+        assertEquals(1, rulesConsequenceEvents.size());
+        final Event rulesConsequenceEvent = rulesConsequenceEvents.get(0);
+        final Map<String, Object> triggeredConsequenceEventData =
+                (Map<String, Object>)
+                        rulesConsequenceEvent.getEventData().get(CONSEQUENCE_TRIGGERED);
+        final Map<String, Object> expectedRuleConsequenceDetails =
+                (Map<String, Object>) expectedRulesConsequenceEventData.get("detail");
+        final Map<String, Object> triggeredRuleConsequenceDetails =
+                (Map<String, Object>) triggeredConsequenceEventData.get("detail");
+        assertEquals(
+                expectedRuleConsequenceDetails.get("schema"),
+                triggeredRuleConsequenceDetails.get("schema"));
+        assertEquals(
+                expectedRuleConsequenceDetails.get("data"),
+                triggeredRuleConsequenceDetails.get("data"));
+        assertEquals(
+                expectedRulesConsequenceEventData.get("type"),
+                triggeredConsequenceEventData.get("type"));
 
+        // clear received events
+        MonitorExtension.reset();
+
+        // trigger the show once in-app message again
+        MobileCore.trackAction("e2e", null);
```

Comment:
the trigger used earlier in this test is `once` but here we're using `e2e`. Are they supposed to be the same?

Reply:
the message triggered by the "e2e" action is a new show once message i had created. the previous message i was testing with was no longer working.

Reply:
Ah ok. Should we use the same trigger on L235 as well then?

Reply:
yeah my bad, this should be using the `once` trigger in both places. will get this fixed.